lib/Chat/ChatManager.php

```diff
@@ -193,11 +185,6 @@ public function waitForNewMessages(Room $chat, $offset, $limit, $timeout, $user)
 		$elapsedTime = 0;
 
 		$comments = $this->commentsManager->getForObjectSince('chat', (string) $chat->getId(), $offset, 'asc', $limit);
-
-		if ($user instanceof IUser) {
-			$this->commentsManager->setReadMark('chat', (string) $chat->getId(), new  \DateTime(), $user);
```

Comment:
Grml, the read marker is set to the time of reading, not the last message timestamp... that's unfortunate and makes things a lot more complicated on the migration :crying_cat_face:

Reply:
We are now setting the message id to -1 on update and if its -1 we get the value on demand and store it.